pkg/controller/constraint/constraint_controller.go

```diff
@@ -629,6 +629,13 @@ func (c *ConstraintsCache) reportTotalConstraints(ctx context.Context, reporter
 }
 
 func IsVapAPIEnabled() (bool, *schema.GroupVersion) {
+	vapMux.RLock()
+	if VapAPIEnabled != nil {
+		vapMux.RUnlock()
```

Comment:
For strict correctness, you should only access the variable while it's under a read lock. Suggest using intermediate variables.

Reply:
Added local variables for strict correctness.